In [1]:
!pip install selenium

In [2]:
import selenium
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.common.exceptions import NoSuchElementException,StaleElementReferenceException
import warnings
warnings.filterwarnings('ignore')
import time

In [3]:
#scraping rest of the data from 'CARDEKHO' website
driver=webdriver.Chrome(r'C:\Users\Admin\Downloads\chromedriver_win32 (1)\chromedriver.exe')
driver.get('https://www.cardekho.com/used-cars+in+ahmedabad')

In [22]:
Brand=[]
for i in driver.find_elements(By.XPATH,'//h3[@class="title"]'):
    Brand.append(i.text.split()[1])
Brand

['Honda',
 'Maruti',
 'Volkswagen',
 'Honda',
 'Honda',
 'Honda',
 'Maruti',
 'Honda',
 'Toyota',
 'Honda',
 'Maruti',
 'Maruti',
 'Hyundai',
 'Honda',
 'Toyota',
 'Mercedes-Benz',
 'Volvo',
 'Porsche',
 'Mercedes-Benz',
 'Mercedes-Benz',
 'Honda',
 'Mahindra',
 'Kia',
 'Maruti',
 'Hyundai',
 'Maruti',
 'Volkswagen',
 'Jeep',
 'Renault',
 'Renault',
 'Nissan',
 'Hyundai',
 'Maruti',
 'Maruti',
 'Renault',
 'Jeep',
 'Volkswagen',
 'Jeep',
 'Renault',
 'Hyundai',
 'Volkswagen',
 'Renault',
 'Maruti',
 'Hyundai',
 'Nissan',
 'Hyundai',
 'Honda',
 'Hyundai',
 'Maruti',
 'Jeep',
 'Maruti',
 'Jeep',
 'Maruti',
 'Hyundai',
 'Hyundai',
 'Honda',
 'Mahindra',
 'Hyundai',
 'Honda',
 'Mahindra',
 'Hyundai',
 'Hyundai',
 'Hyundai',
 'Hyundai',
 'Maruti',
 'Renault',
 'Hyundai',
 'Hyundai',
 'Hyundai',
 'Jaguar',
 'Maruti',
 'Hyundai',
 'Toyota',
 'Maruti',
 'Kia',
 'BMW',
 'BMW',
 'Hyundai',
 'Renault',
 'Hyundai',
 'BMW',
 'Honda',
 'Hyundai',
 'Maruti',
 'Honda',
 'BMW',
 'Volkswagen',
 'Hyundai

In [4]:
name=[]
for i in driver.find_elements(By.XPATH,'//h3[@class="title"]'):
    name.append(i.text)
name

['2020 Honda City V MT',
 '2017 Maruti Baleno 1.2 Alpha',
 '2016 Volkswagen Polo 1.2 MPI Highline',
 '2020 Honda City ZX CVT',
 '2016 Honda Jazz V CVT',
 '2014 Honda City i VTEC CVT SV',
 '2017 Maruti Ciaz 1.4 Delta',
 '2017 Honda City i-VTEC CVT ZX',
 '2016 Toyota Camry 2.5 G',
 '2018 Honda Jazz VX CVT',
 '2019 Maruti Baleno 1.2 Zeta',
 '2018 Maruti Alto 800 LXI',
 '2017 Hyundai i20 1.4 Asta',
 '2021 Honda WR-V VX Diesel',
 '2021 Toyota Fortuner Legender',
 '2015 Mercedes-Benz E-Class E400 Cabriolet Edition E',
 '2019 Volvo XC 90 D5 Momentum',
 '2017 Porsche Macan R4',
 '2019 Mercedes-Benz E-Class Exclusive E 220d',
 '2021 Mercedes-Benz GLS 400d 4MATIC',
 '2016 Honda Jazz 1.2 V AT i VTEC',
 '2020 Mahindra XUV300 W6',
 '2019 Kia Seltos GTX Plus',
 '2018 Maruti Swift Dzire VXI 1.2',
 '2016 Hyundai Grand i10 AT Asta',
 '2016 Maruti Ciaz VXi Plus',
 '2017 Volkswagen Vento 1.2 TSI Highline AT',
 '2019 Jeep Compass 1.4 Limited Plus BSIV',
 '2021 Renault Duster RXZ Turbo CVT',
 '2018 Renault

In [20]:
name=[s[5:] for s in name]
name

['Honda City V MT',
 'Maruti Baleno 1.2 Alpha',
 'Volkswagen Polo 1.2 MPI Highline',
 'Honda City ZX CVT',
 'Honda Jazz V CVT',
 'Honda City i VTEC CVT SV',
 'Maruti Ciaz 1.4 Delta',
 'Honda City i-VTEC CVT ZX',
 'Toyota Camry 2.5 G',
 'Honda Jazz VX CVT',
 'Maruti Baleno 1.2 Zeta',
 'Maruti Alto 800 LXI',
 'Hyundai i20 1.4 Asta',
 'Honda WR-V VX Diesel',
 'Toyota Fortuner Legender',
 'Mercedes-Benz E-Class E400 Cabriolet Edition E',
 'Volvo XC 90 D5 Momentum',
 'Porsche Macan R4',
 'Mercedes-Benz E-Class Exclusive E 220d',
 'Mercedes-Benz GLS 400d 4MATIC',
 'Honda Jazz 1.2 V AT i VTEC',
 'Mahindra XUV300 W6',
 'Kia Seltos GTX Plus',
 'Maruti Swift Dzire VXI 1.2',
 'Hyundai Grand i10 AT Asta',
 'Maruti Ciaz VXi Plus',
 'Volkswagen Vento 1.2 TSI Highline AT',
 'Jeep Compass 1.4 Limited Plus BSIV',
 'Renault Duster RXZ Turbo CVT',
 'Renault KWID RXL',
 'Nissan Micra XL Optional',
 'Hyundai Grand i10 Magna',
 'Maruti Swift Dzire VXI',
 'Maruti Wagon R LXI BS IV',
 'Renault KWID RXL',
 'Je

In [8]:
len(name)

140

In [9]:
manufacturing_year=[]
for i in driver.find_elements(By.XPATH,'//h3[@class="title"]'):
    manufacturing_year.append(i.text.split()[0])
manufacturing_year

['2020',
 '2017',
 '2016',
 '2020',
 '2016',
 '2014',
 '2017',
 '2017',
 '2016',
 '2018',
 '2019',
 '2018',
 '2017',
 '2021',
 '2021',
 '2015',
 '2019',
 '2017',
 '2019',
 '2021',
 '2016',
 '2020',
 '2019',
 '2018',
 '2016',
 '2016',
 '2017',
 '2019',
 '2021',
 '2018',
 '2015',
 '2016',
 '2014',
 '2014',
 '2021',
 '2018',
 '2017',
 '2017',
 '2019',
 '2019',
 '2015',
 '2017',
 '2015',
 '2018',
 '2021',
 '2018',
 '2015',
 '2021',
 '2014',
 '2017',
 '2018',
 '2019',
 '2018',
 '2018',
 '2016',
 '2016',
 '2017',
 '2015',
 '2015',
 '2016',
 '2017',
 '2016',
 '2014',
 '2014',
 '2016',
 '2016',
 '2016',
 '2019',
 '2013',
 '2012',
 '2017',
 '2020',
 '2020',
 '2017',
 '2021',
 '2012',
 '2013',
 '2011',
 '2018',
 '2009',
 '2014',
 '2011',
 '2020',
 '2012',
 '2013',
 '2019',
 '2019',
 '2012',
 '2015',
 '2020',
 '2015',
 '2017',
 '2013',
 '2017',
 '2014',
 '2020',
 '2018',
 '2016',
 '2018',
 '2012',
 '2014',
 '2015',
 '2021',
 '2004',
 '2017',
 '2011',
 '2015',
 '2019',
 '2014',
 '2018',
 '2021',
 

In [11]:
driven_kilometers1=[]
for i in driver.find_elements(By.XPATH,'//div[@class="dotsDetails"]'):
    driven_kilometers1.append(i.text.split()[0])
driven_kilometers1

['38,906',
 '38,203',
 '69,414',
 '21,261',
 '39,821',
 '1,10,284',
 '34,971',
 '92,246',
 '60,343',
 '26,696',
 '20,548',
 '21,186',
 '85,672',
 '14,470',
 '6,500',
 '45,000',
 '55,000',
 '25,000',
 '35,000',
 '1,100',
 '15,385',
 '10,381',
 '18,557',
 '50,215',
 '59,313',
 '36,534',
 '49,719',
 '32,378',
 '13,248',
 '34,972',
 '35,894',
 '34,438',
 '56,568',
 '90,244',
 '4,491',
 '78,151',
 '43,688',
 '86,226',
 '17,346',
 '23,837',
 '71,607',
 '48,872',
 '41,329',
 '21,239',
 '5,717',
 '66,375',
 '46,623',
 '11,945',
 '62,361',
 '58,463',
 '51,324',
 '15,414',
 '64,303',
 '52,504',
 '87,115',
 '46,417',
 '52,514',
 '47,431',
 '1,00,161',
 '85,443',
 '51,085',
 '35,424',
 '57,581',
 '42,593',
 '30,983',
 '32,681',
 '56,285',
 '39,000',
 '1,20,000',
 '15,000',
 '1,00,000',
 '14,450',
 '37,001',
 '1,20,000',
 '13,000',
 '40,000',
 '16,000',
 '65,000',
 '31,000',
 '51,000',
 '34,000',
 '89,000',
 '44,001',
 '93,000',
 '85,000',
 '49,000',
 '33,000',
 '55,000',
 '60,000',
 '58,000',
 '66

In [16]:
Fuel1=[]
for i in driver.find_elements(By.XPATH,'//div[@class="dotsDetails"]'):
    Fuel1.append(i.text.split()[3])
Fuel1

['Petrol',
 'Petrol',
 'Petrol',
 'Petrol',
 'Petrol',
 'Petrol',
 'Petrol',
 'Petrol',
 'Petrol',
 'Petrol',
 'Petrol',
 'Petrol',
 'Diesel',
 'Diesel',
 'Diesel',
 'Petrol',
 'Diesel',
 'Petrol',
 'Diesel',
 'Diesel',
 'Petrol',
 'Petrol',
 'Petrol',
 'Petrol',
 'Petrol',
 'Petrol',
 'Petrol',
 'Petrol',
 'Petrol',
 'Petrol',
 'Petrol',
 'Petrol',
 'Petrol',
 'Petrol',
 'Petrol',
 'Petrol',
 'Petrol',
 'Diesel',
 'Petrol',
 'Petrol',
 'Diesel',
 'Petrol',
 'Petrol',
 'Petrol',
 'Petrol',
 'Petrol',
 'Petrol',
 'Petrol',
 'Petrol',
 'Diesel',
 'Petrol',
 'Diesel',
 'Petrol',
 'Petrol',
 'Diesel',
 'Petrol',
 'Diesel',
 'Petrol',
 'Petrol',
 'Diesel',
 'Petrol',
 'Petrol',
 'Petrol',
 'Petrol',
 'Petrol',
 'Petrol',
 'Diesel',
 'Petrol',
 'Diesel',
 'Diesel',
 'Diesel',
 'Petrol',
 'Diesel',
 'Diesel',
 'Petrol',
 'Diesel',
 'Diesel',
 'Petrol',
 'Diesel',
 'Petrol',
 'Diesel',
 'Petrol',
 'Diesel',
 'Diesel',
 'Petrol',
 'Diesel',
 'Petrol',
 'Petrol',
 'Diesel',
 'Petrol',
 'Petrol',

In [17]:
Price1=[]
for i in driver.find_elements(By.XPATH,'//div[@class="Price hover"]'):
    Price1.append(i.text)
Price1

['₹ 10.03 Lakh',
 '₹ 6.43 Lakh',
 '₹ 5.15 Lakh',
 '₹ 12.99 Lakh',
 '₹ 6.05 Lakh',
 '₹ 5.43 Lakh',
 '₹ 6.73 Lakh',
 '₹ 9.05 Lakh',
 '₹ 16.40 Lakh',
 '₹ 7.77 Lakh',
 '₹ 6.61 Lakh',
 '₹ 3.23 Lakh',
 '₹ 6.67 Lakh',
 '₹ 11.60 Lakh',
 '₹ 48.50 Lakh',
 '₹ 48.50 Lakh',
 '₹ 58.00 Lakh',
 '₹ 58.00 Lakh',
 '₹ 53.00 Lakh',
 '₹ 1.36 Cr',
 '₹ 6.00 Lakh',
 '₹ 8.62 Lakh',
 '₹ 15.53 Lakh',
 '₹ 5.94 Lakh',
 '₹ 5.63 Lakh',
 '₹ 6.12 Lakh',
 '₹ 7.66 Lakh',
 '₹ 16.78 Lakh',
 '₹ 12.83 Lakh',
 '₹ 3.26 Lakh',
 '₹ 3.21 Lakh',
 '₹ 4.43 Lakh',
 '₹ 4.34 Lakh',
 '₹ 3.12 Lakh',
 '₹ 4.13 Lakh',
 '₹ 11.90 Lakh',
 '₹ 7.58 Lakh',
 '₹ 10.03 Lakh',
 '₹ 3.70 Lakh',
 '₹ 5.60 Lakh',
 '₹ 5.11 Lakh',
 '₹ 3.11 Lakh',
 '₹ 2.45 Lakh',
 '₹ 5.41 Lakh',
 '₹ 7.44 Lakh',
 '₹ 8.60 Lakh',
 '₹ 4.21 Lakh',
 '₹ 22.78 Lakh',
 '₹ 2.23 Lakh',
 '₹ 12.43 Lakh',
 '₹ 6.12 Lakh',
 '₹ 15.88 Lakh',
 '₹ 7.53 Lakh',
 '₹ 6.88 Lakh',
 '₹ 5.83 Lakh',
 '₹ 8.39 Lakh',
 '₹ 9.50 Lakh',
 '₹ 4.93 Lakh',
 '₹ 4.20 Lakh',
 '₹ 10.09 Lakh',
 '₹ 2.93 Lakh',
 '₹ 4.91

In [26]:
product_urls1 = []
url=driver.find_elements(By.XPATH,'//h3[@class="title"]//a')   
for i in url:
    product_urls1.append(i.get_attribute('href'))                  
product_urls1

['https://www.cardekho.com/buy-used-car-details/used-Honda-City-V-Mt-cars-Ahmedabad_8a3ef5f4-820c-4994-b188-2dd827211849.htm',
 'https://www.cardekho.com/buy-used-car-details/used-Maruti-Baleno-1.2-Alpha-cars-Ahmedabad_31684dec-0513-449a-b821-97895e1b2295.htm',
 'https://www.cardekho.com/buy-used-car-details/used-Volkswagen-Polo-1.2-Mpi-Highline-cars-Ahmedabad_f475eec4-a9ca-4cb3-87ed-f55443375de7.htm',
 'https://www.cardekho.com/buy-used-car-details/used-Honda-City-Zx-Cvt-cars-Ahmedabad_38d158d8-9fb2-4418-bd76-ea80c4658363.htm',
 'https://www.cardekho.com/buy-used-car-details/used-Honda-Jazz-V-Cvt-cars-Ahmedabad_c335050b-dda0-45a6-a135-54b26b902f88.htm',
 'https://www.cardekho.com/buy-used-car-details/used-Honda-City-I-Vtec-Cvt-Sv-cars-Ahmedabad_5bd6ddf7-e284-417c-941c-b88be68ebc2d.htm',
 'https://www.cardekho.com/buy-used-car-details/used-Maruti-Ciaz-1.4-Delta-cars-Ahmedabad_22930b6d-9974-468d-b3fb-f0a43328c6fb.htm',
 'https://www.cardekho.com/buy-used-car-details/used-Honda-City-I-vt

In [36]:
Location1 = []   
for i in product_urls1:
    driver.get(i)
    time.sleep(1)
    try:
        ow=driver.find_element(By.XPATH,'//span[@class="value-text truncate-value hover posR"]')
        Location1.append(ow.text.split()[0])
    except NoSuchElementException:
        Location1.append('-')
Location1

['Ahmedabad,',
 'Ahmedabad,',
 'Ahmedabad,',
 'Ahmedabad,',
 'Ahmedabad,',
 'Ahmedabad,',
 'Ahmedabad,',
 'Ahmedabad,',
 'Ahmedabad,',
 'Ahmedabad,',
 'Ahmedabad,',
 'Ahmedabad,',
 'Ahmedabad,',
 'Ahmedabad,',
 '-',
 '-',
 '-',
 '-',
 '-',
 '-',
 'Ahmedabad,',
 'Ahmedabad,',
 'Ahmedabad,',
 'Ahmedabad,',
 'Ahmedabad,',
 'Ahmedabad,',
 'Ahmedabad,',
 'Ahmedabad,',
 'Ahmedabad,',
 'Ahmedabad,',
 'Ahmedabad,',
 'Ahmedabad,',
 'Ahmedabad,',
 'Ahmedabad,',
 'Ahmedabad,',
 'Ahmedabad,',
 'Ahmedabad,',
 'Ahmedabad,',
 'Ahmedabad,',
 'Ahmedabad,',
 'Ahmedabad,',
 'Ahmedabad,',
 'Ahmedabad,',
 'Ahmedabad,',
 'Ahmedabad,',
 'Ahmedabad,',
 'Ahmedabad,',
 'Ahmedabad,',
 'Ahmedabad,',
 'Ahmedabad,',
 'Ahmedabad,',
 'Ahmedabad,',
 'Ahmedabad,',
 'Ahmedabad,',
 'Ahmedabad,',
 'Ahmedabad,',
 'Ahmedabad,',
 'Ahmedabad,',
 'Ahmedabad,',
 'Ahmedabad,',
 'Ahmedabad,',
 'Ahmedabad,',
 'Ahmedabad,',
 'Ahmedabad,',
 'Ahmedabad,',
 'Ahmedabad,',
 'Ahmedabad,',
 '-',
 '-',
 '-',
 '-',
 '-',
 '-',
 '-',
 '-',
 

In [42]:
No_of_Owner1 = []   
for i in product_urls1:
    driver.get(i)
    time.sleep(1)
    try:
        ow=driver.find_element(By.XPATH,'//ul[@class="gsc_row detailsList"]')
        No_of_Owner1.append(ow.text.split('\n')[13])
    except NoSuchElementException:
        No_of_Owner1.append('-')
No_of_Owner1        

['-',
 'First Owner',
 '-',
 '-',
 '-',
 '-',
 '-',
 '-',
 'First Owner',
 '-',
 '-',
 '-',
 '-',
 '-',
 'First Owner',
 'First Owner',
 'First Owner',
 'First Owner',
 'First Owner',
 'First Owner',
 'First Owner',
 '-',
 'First Owner',
 '-',
 '-',
 '-',
 '-',
 '-',
 '-',
 '-',
 '-',
 '-',
 '-',
 'First Owner',
 '-',
 '-',
 '-',
 'First Owner',
 '-',
 '-',
 '-',
 '-',
 'First Owner',
 '-',
 '-',
 '-',
 '-',
 '-',
 '-',
 'First Owner',
 '-',
 '-',
 '-',
 'First Owner',
 '-',
 '-',
 '-',
 '-',
 '-',
 '-',
 '-',
 '-',
 '-',
 '-',
 'First Owner',
 '-',
 '-',
 'First Owner',
 'Manual',
 'First Owner',
 'Manual',
 'First Owner',
 'First Owner',
 'Manual',
 'First Owner',
 'First Owner',
 'First Owner',
 'First Owner',
 'First Owner',
 'First Owner',
 'First Owner',
 'First Owner',
 'First Owner',
 'First Owner',
 'First Owner',
 'First Owner',
 'First Owner',
 'First Owner',
 'First Owner',
 'First Owner',
 'First Owner',
 'First Owner',
 'First Owner',
 'First Owner',
 'First Owner',
 'Fir

In [43]:
print(len(Brand),len(name),len(manufacturing_year),len(driven_kilometers1),len(Fuel1),len(Price1),len(Location1),len(No_of_Owner1),len(product_urls1))

140 140 140 140 140 140 140 140 140


In [44]:
import pandas as pd
df=pd.DataFrame({'Brand':Brand,'Car Name':name,'Manufacturing Year':manufacturing_year,'Driven KMs':driven_kilometers1,'Fuel Type':Fuel1,'Car Price':Price1,'Location':Location1,'Total Owners':No_of_Owner1,'URL"s':product_urls1})
df

Brand                          Car Name Manufacturing Year  \
0         Honda                   Honda City V MT               2020   
1        Maruti           Maruti Baleno 1.2 Alpha               2017   
2    Volkswagen  Volkswagen Polo 1.2 MPI Highline               2016   
3         Honda                 Honda City ZX CVT               2020   
4         Honda                  Honda Jazz V CVT               2016   
..          ...                               ...                ...   
135      Maruti               Maruti Alto 800 LXI               2020   
136      Maruti         Maruti Omni 8 Seater BSII               2008   
137     Hyundai  Hyundai Creta 1.6 CRDi SX Option               2017   
138      Maruti             Maruti Swift LXI BSIV               2015   
139      Maruti              Maruti Ciaz VXi Plus               2016   

    Driven KMs Fuel Type     Car Price    Location Total Owners  \
0       38,906    Petrol  ₹ 10.03 Lakh  Ahmedabad,            -   
1       38,203    Petrol   ₹ 6.43 Lakh  Ahmedabad,  First Owner   
2       69,414    Petrol   ₹ 5.15 Lakh  Ahmedabad,            -   
3       21,261    Petrol  ₹ 12.99 Lakh  Ahmedabad,            -   
4       39,821    Petrol   ₹ 6.05 Lakh  Ahmedabad,            -   
..         ...       ...           ...         ...          ...   
135     25,000    Petrol   ₹ 3.40 Lakh           -       Manual   
136     15,000    Petrol       ₹ 48000           -       Manual   
137     93,409    Diesel   ₹ 9.12 Lakh           -  First Owner   
138     53,123    Petrol   ₹ 3.64 Lakh           -  First Owner   
139     63,628    Petrol   ₹ 6.15 Lakh           -  First Owner   

                                                 URL"s  
0    https://www.cardekho.com/buy-used-car-details/...  
1    https://www.cardekho.com/buy-used-car-details/...  
2    https://www.cardekho.com/buy-used-car-details/...  
3    https://www.cardekho.com/buy-used-car-details/...  
4    https://www.cardekho.com/buy-used-car-details/...  
..                                                 ...  
135  https://www.cardekho.com/used-car-details/used...  
136  https://www.cardekho.com/used-car-details/used...  
137  https://www.cardekho.com/used-car-details/used...  
138  https://www.cardekho.com/used-car-details/used...  
139  https://www.cardekho.com/used-car-details/used...  

[140 rows x 9 columns]

In [45]:
df.to_excel('Cardekho_used_car.xlsx')

##### In this notebook i've found-out 140 used car information from Cardekho website